File: ecommerce_app/ecomerce_app_frontend/lib/models/product.dart
class Product {
  final String id;
  final String name;
  final String description;
  final double price;
  final List<String> images;
  final List<String> variants;
  final double rating;

  Product({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.images,
    required this.variants,
    required this.rating,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['_id'],
      name: json['name'],
      description: json['description'],
      price: json['price'].toDouble(),
      images: List<String>.from(json['images']),
      variants: List<String>.from(json['variants']),
      rating: json['rating'].toDouble(),
    );
  }
}

File: ecommerce_app/ecomerce_app_backend/models/product.js
const mongoose = require('mongoose');

// Định nghĩa schema cho sản phẩm
const productSchema = new mongoose.Schema({
  name: { type: String, required: true },
  description: { type: String, required: true },
  price: { type: Number, required: true },
  category: { type: String, required: true },
  images: [String],    // Mảng chứa URL ảnh
  variants: [String],  // Các biến thể của sản phẩm (màu sắc, kích thước, v.v.)
  rating: { type: Number, min: 1, max: 5, default: 1 },
});

// Tạo mô hình sản phẩm từ schema
module.exports = mongoose.model('Product', productSchema);
